demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -105,108 +143,179 @@ def convert_epoch_time_to_string_time(
+
+    Returns:
+        str: The pretty name of the type of the given object.
+    """
+    return {
+        type(None): "null",
+        list: "list",
+        dict: "dict",
+        tuple: "tuple",
+        set: "set",
+        UUID: "UUID",
+        str: "string",
+        int: "int",
+        float: "float",
+        bool: "boolean",
+        datetime: "datetime",
+    }.get(type(obj), str(type(obj)))
 
 
 def verify_results(
+    modeling_rule: ModelingRule,
     tested_dataset: str,
     results: List[dict],
     test_data: init_test_data.TestData,
-):
+) -> List[TestCase]:
     """Verify that the results of the XQL query match the expected values.
 
     Args:
+        modeling_rule (ModelingRule): The modeling rule object parsed from the modeling rule file.
         tested_dataset (str): The dataset to verify result for.
         results (List[dict]): The results of the XQL query.
         test_data (init_test_data.TestData): The data parsed from the test data file.
 
     Returns:
-        bool: True if the results are valid, False otherwise.
+        list[TestCase]: List of test cases for the results of the XQL query.
     """
+
     if not results:
-        err = (
-            "[red]No results were returned by the query - it's possible there is a syntax"
-            " error with your modeling rule and that it did not install properly on the tenant[/red]"
+        logger.error(
+            f"[red]{SYNTAX_ERROR_IN_MODELING_RULE}[/red]", extra={"markup": True}
+        )
+        test_case = TestCase(
+            f"Modeling rule - {modeling_rule.normalize_file_name()}",
+            classname="Modeling Rule Results",
         )
-        logger.error(err, extra={"markup": True})
+        test_case.result += [Failure(SYNTAX_ERROR_IN_MODELING_RULE)]
+        test_case.system_err = SYNTAX_ERROR_IN_MODELING_RULE
+        return [test_case]
 
-        return False
     rule_relevant_data = [
         data for data in test_data.data if data.dataset == tested_dataset
     ]
     if len(results) != len(rule_relevant_data):
         err = (
-            f"[red]Expected {len(test_data.data)} results, got {len(results)}. Verify that the event"
+            f"Expected {len(test_data.data)} results, got {len(results)}. Verify that the event"
             " data used in your test data file meets the criteria of the modeling rule, e.g. the filter"
-            " condition.[/red]"
+            " condition."
         )
-        logger.error(err, extra={"markup": True})
-        return False
-    errors = False
+        test_case = TestCase(
+            f"Modeling rule - {modeling_rule.normalize_file_name()}",
+            classname="Modeling Rule Results",
+        )
+        test_case.result += [Failure(err)]
+        logger.error(f"[red]{err}[/red]", extra={"markup": True})
+        return [test_case]
 
-    for i, result in enumerate(results):
+    test_cases = []
+    for i, result in enumerate(results, start=1):
```

Comment:
Please consider extracting the code in the loop to another function.

Reply:
done